lib/application.js

```diff
@@ -618,6 +618,34 @@ app.listen = function listen() {
   return server.listen.apply(server, arguments);
 };
 
+/**
+ * Creates a router that inherits options from app.settings
+ * and allows them to be overriden by passing in an options
+ * object
+ *
+ * @param {Object} [options]
+ * @return {router}
+ * @public
+ */
+app.createRouter = function (options) {
+  var opts = {
+    strict: this.enabled('strict routing'),
+    caseSensitive: this.enabled('case sensitive routing')
+  }
+
+  if (options !== undefined) {
+    if (options.caseSensitive !== undefined) {
+      opts.caseSensitive = options.caseSensitive
+    }
+
+    if (options.strict !== undefined) {
+      opts.strict = options.strict
+    }
+  }
+
+  return Router(opts)
+}
+
```

Comment:
This is mostly just a style thing, so if there is strong opposition feel free to ignore me, but I find that reducing the code required to achieve the same result often reduces maintenance and understanding costs in the future.  For example, this could be written much more succinctly as follows:

```javascript
app.createRouter = function (options) {
  var opts = options || {}
  opts.strict = options.strict !== undefined ? options.strict : this.enabled('strict routing')
  opts.caseSensitive = options.caseSensitive !== undefined ? options.caseSensitive : this.enabled('case sensitive routing')
  return Router(opts)
}
```

Sometimes more lines of code is a good thing for clarity, but in this case, there are a bunch of lines which can be achieved with one usage of a ternary.  This also has the upside of not unnecessarily calling `this.enabled` if you pass the option.

Reply:
I like that we don't have to call this.enabled if the options are passed in, that's good. But to me this looks like a jumbled mess. It takes multiple seconds to decode what it actually means while undefined checks are really easy to make sense of imo.

Reply:
Like I said, mostly a style thing :), but maybe you can find a middle ground without the unnecessary calls?

Also, technically it is just the same `undefined` checks you had but in a single line.  Maybe other people like the more verbose syntax with the same result, I don't know and it probably shouldn't be the focus of any more energy.